logrus.go

```diff
@@ -6,69 +6,85 @@ import (
 	"os"
 	"time"
 
-	"go.mozilla.org/mozlog"
-
 	"github.com/sirupsen/logrus"
 )
 
-// Enable enables mozlogrus
-func Enable(loggerName string) {
-	logrus.SetFormatter(&MozLogFormatter{
-		LoggerName: loggerName,
-	})
+var hostname string
+var pid int
 
-	logrus.SetOutput(os.Stdout)
-}
+func init() {
+	var err error
+	hostname, err = os.Hostname()
+	if err != nil {
+		hostname = "unknown"
+	}
 
-// SetFormatterName the LoggerName for the logrus formatter
-func SetFormatterName(name string) {
-	logrus.SetFormatter(&MozLogFormatter{
-		LoggerName: name,
-	})
+	pid = os.Getpid()
 }
 
-// prefixFieldClashes is from logrus package
-func prefixFieldClashes(data logrus.Fields) {
-	_, ok := data["msg"]
-	if ok {
-		data["fields.msg"] = data["msg"]
-	}
+// Enable changes the default logrus formatter to MozLogFormatter and
+// sets output to stdout
+func Enable(m *MozLogFormatter) {
+	logrus.SetFormatter(m)
+	logrus.SetOutput(os.Stdout)
 }
 
 type MozLogFormatter struct {
-	LoggerName string
+	Name string
+	Type string
 }
 
 func (m *MozLogFormatter) Format(entry *logrus.Entry) ([]byte, error) {
-	appLog := &mozlog.AppLog{
-		Timestamp:  entry.Time.UTC().UnixNano(),
-		Time:       entry.Time.UTC().Format(time.RFC3339),
-		Type:       "app.log",
-		Logger:     m.LoggerName,
-		Hostname:   mozlog.Hostname(),
+	t := entry.Time.UTC()
+	appLog := &appLog{
+		Timestamp:  t.UnixNano(),
+		Time:       t.Format(time.RFC3339),
+		Type:       m.Type,
+		Logger:     m.Name,
+		Hostname:   hostname,
 		EnvVersion: "2.0",
-		Pid:        os.Getpid(),
-		Severity:   int(entry.Level),
+		Pid:        pid,
+		Severity:   toSyslogSeverity(entry.Level),
 	}
 
-	data := make(logrus.Fields, len(entry.Data)+1)
```

Comment:
I'm kind of neutral on this, but doing this adds a side effect in the case where entry is reused for some reason.

Reply:
I don't think it's worth the resource cost to duplicate the `map` in case somebody reuses an `Entry`.  Most of the idiomatic code/examples seem to use this pattern: `logrus.WithFields(fields).Info("message")` where the `Entry` is immediately discarded.

Reply:
That's fine. The side effect should be noted in the method comment.